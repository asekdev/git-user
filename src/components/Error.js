import React from "react";

const Error = ({ error }) => {
	return (
		<div>
			<pre>{error}</pre>
		</div>
	);
};

export default Error;
