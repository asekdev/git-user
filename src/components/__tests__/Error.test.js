import React from "react";
import renderer from "react-test-renderer";
import Error from "../Error";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

test("testing error", () => {
  const component = renderer.create(<Error error="Found an error!" />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("error prop gets passed successfully", () => {
  const { getByText } = render(<Error error="Found an error!" />);
  expect(getByText(/Found an error!/)).toBeTruthy();
});

test("error prop gets passed successfully", () => {
  const { getByTestId } = render(<Error error="Found an error!" />);
  expect(getByTestId("error-text")).toBeTruthy();
});
