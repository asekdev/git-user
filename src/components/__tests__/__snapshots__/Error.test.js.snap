// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing error 1`] = `
<div
  className="makeStyles-container-1"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root makeStyles-icon-2"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm1 13h-2v-2h2v2zm0-4h-2V7h2v6z"
      opacity=".3"
    />
    <path
      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm-1-5h2v2h-2zm0-8h2v6h-2z"
    />
  </svg>
  <h5
    className="MuiTypography-root makeStyles-text-3 MuiTypography-h5 MuiTypography-gutterBottom"
    data-testid="error-text"
  >
    Found an error!
  </h5>
</div>
`;
